### ISSUE TITLE
Modal: Add headerLevel prop for modal header.

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#1977]
[A11Y-603](https://jira.cengage.com/browse/A11Y-603)
[DBSFE-7543](https://jira.cengage.com/browse/DBSFE-7543)

Now, we have a default `level=1` for the modal header.
The goal of this task is to add a new prop `headerLevel` to add the possibility to update the level of the header.
Update the default header level from `H1` to `H2`

Original request: https://cengage.slack.com/archives/C8Y4FR63U/p1760972965446949

In most cases, we should use the `H2` heading for Modal. 
https://mui.com/material-ui/react-modal/
https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/#:~:text=%3E%0A%C2%A0%C2%A0%C2%A0%C2%A0%3C-,h2,-id%3D%22dialog1_label

**Comment by nikitaorliak-cengage** [comment]
Works as expected.
